data/lightdm-mini-greeter.conf

```diff
@@ -53,10 +53,12 @@ text-color = "#080800"
 # The color of the error text
 error-color = "#F8F8F0"
 # An absolute path to an optional background image.
-# The image will be displayed centered & unscaled.
 # Note: The file should be somewhere that LightDM has permissions to read
-#       (e.g., /etc/lightdm/).
+# (e.g., /etc/lightdm/).
 background-image = ""
+# Background image size: cover, contain
```

Comment:
Is `auto` valid as well? Which setting retains the current image rendering? 

When adding settings, I try & keep the default to whatever maintains previous behaviour.

Reply:
Basically all options listed [here](https://www.w3.org/TR/css-backgrounds-3/#background-size) working properly and is valid to use with introduced option.

Previous behaviour was `auto` mode, which just takes image as is and place it in the center. So if your screen resolution is 1920x1080 and image is 1024x720, this image will be same size with no scaling with black frames around. And if image is larger then screen res, then it will be cropped.

With scaling mode you can do a lot more, i.e:

`background-image-size = contain` will scale image up (or down) until it's width or height reaches the max screen size from inside (preserving image proportions). But if your monitor have 16x10 aspect ratio, and image has 16x9 you'll still get small black frames on top and bottom.

So here comes `background-image-size = cover` which will scale image up (or down) until it's smallest size will reach max width or height of the screen, completely covering all screen surface. The rest will be cropped from sides. Which is the best resulting mode for wallpaper purpose.

Also if needed you can adjust the percentage like this: `background-image-size = 100% 100%` - in this case 16x9 image will be stretched to 16x10 monitor without cropping, but breaking it's natural aspect ratio. You can set any percentage here, which are relative to screen width and height. Also you can mix percentage and `auto` value to preserve image aspect ratio, i.e: `background-image-size = 150% auto` will zoom in 1,5 times with proper image proportions.

** Going further, it's also possible to introduce image `fill` option for background, i.e if you have some tile image 50x50 and you need to fill all background space with this image. For this feature some more patching need to be done with introducing new option `background-image-repeat = true|false`. If you feel that this will be useful, I can update PR.